File: task-scheduler-frontend/src/types/notification.ts
export type NotificationType = 'TASK_ASSIGNED' | 'TASK_REASSIGNED' | 'TASK_COMPLETED' | 'TASK_OVERDUE' | 'COMMENT_MENTION';

export interface BackendNotification {
  id: string;
  userId: string;
  type: string;
  referenceType: string;
  referenceId: string;
  message: string;
  isRead: boolean;
  createdAt: string;
  projectId?: string;
  senderId?: string;
  action: string;
  metadata: any;
}

// Interface for UI components - camelCase for frontend use
export interface Notification {
  id: string;
  userId: string;
  title: string;
  message: string;
  type: NotificationType;
  read: boolean;
  createdAt: string;
  projectId?: string;
  taskId?: string;
  commentId?: string;
  senderId?: string;
  link?: string;
}

export interface NotificationState {
  notifications: Notification[];
  unreadCount: number;
}

export interface NotificationCount {
  total: number;
  unread: number;
}

export interface CreateNotificationInput {
  userId: string;
  title: string;
  message: string;
  type: NotificationType;
  projectId?: string;
  taskId?: string;
  commentId?: string;
  senderId?: string;
}

export interface NotificationQueryResponse {
  notifications: BackendNotification[];
}

export interface NotificationCountQueryResponse {
  notificationCount: NotificationCount;
}

File: task-scheduler-backend/src/db/models/notification.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::types::Uuid;
use sqlx::types::JsonValue;

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct Notification {
    pub notification_id: Uuid,
    pub user_id: Uuid,
    pub project_id: Option<Uuid>,
    pub sender_id: Option<Uuid>,
    pub type_: String,
    pub reference_type: String,
    pub reference_id: Uuid,
    pub message: String,
    pub action: String,
    pub metadata: JsonValue,
    pub is_read: bool,
    pub created_at: DateTime<Utc>,
    pub read_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateNotificationInput {
    pub user_id: Uuid,
    pub project_id: Option<Uuid>,
    pub sender_id: Option<Uuid>,
    pub type_: String,
    pub reference_type: String,
    pub reference_id: Uuid,
    pub message: String,
    pub action: String,
    pub metadata: Option<JsonValue>,
} 